Data/models/Device.swift

```diff
@@ -49,12 +51,25 @@ public final class Device: NSManagedObject, Syncable, CRUD {
         
         // No guard, let bleed through to allow 'empty' devices (e.g. local)
         let root = root as? SyncDevice
-        let device = Device(entity: Device.entity(context: context), insertInto: context)
         
-        device.created = root?.syncNativeTimestamp ?? Date()
-        device.syncUUID = root?.objectId ?? SyncCrypto.uniqueSerialBytes(count: 16)
+        let syncUUID = root?.objectId ?? SyncCrypto.uniqueSerialBytes(count: 16)
         
-        device.update(syncRecord: root)
+        var device: Device?
+        if let syncDisplayUUID = SyncHelpers.syncDisplay(fromUUID: syncUUID) {
+            // There can't be more than two device with the same syncUUID.
+            // A race condition could sometimes happen while getting two `Sync.resolvedSyncRecords` callbacks at the same time.
+            // Fixing issue #692 should help with it, until then we do a simple guard and disallow adding another device with the same syncUUID.
+            let predicate = NSPredicate(format: "syncDisplayUUID == %@", syncDisplayUUID)
+            device = Device.first(where: predicate, context: context)
+        }
+        
+        if device == nil {
+            device = Device(entity: Device.entity(context: context), insertInto: context)
+            device?.created = root?.syncNativeTimestamp ?? Date()
+            device?.syncUUID = syncUUID
+            
+            device?.update(syncRecord: root)
```

Comment:
I think we can place this outside the `if`? That way it would update regardless of what device it will use.